From eee84c642fc49a05871fc74aca84e106e05a1b14 Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Sat, 16 Jul 2011 13:41:40 +0200
Subject: [PATCH 01/11] added BB_SVN_TUNNEL variable

It allows to rewrite svn:// uris for example to svn+proxy://.  This
makes it possible to use a 'proxy' tunnel which pipes the connection
through a https proxy.
---
 lib/bb/fetch2/svn.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/bb/fetch2/svn.py b/lib/bb/fetch2/svn.py
index 8a29193..2820a32 100644
--- a/lib/bb/fetch2/svn.py
+++ b/lib/bb/fetch2/svn.py
@@ -51,6 +51,7 @@ class Svn(FetchMethod):
             raise MissingParameterError('module', ud.url)
 
         ud.basecmd = d.getVar('FETCHCMD_svn', True)
+        ud.tunnel  = data.getVar('BB_SVN_TUNNEL', d, True) or ""
 
         ud.module = ud.parm["module"]
 
@@ -79,6 +80,9 @@ class Svn(FetchMethod):
 
         proto = ud.parm.get('protocol', 'svn')
 
+        if proto == "svn":
+            proto = proto + ud.tunnel
+
         svn_rsh = None
         if proto == "svn+ssh" and "rsh" in ud.parm:
             svn_rsh = ud.parm["rsh"]
-- 
2.5.5

